src/State/Processor/RespondProcessor.php

```diff
@@ -64,6 +66,11 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             'X-Frame-Options' => 'deny',
         ];
 
+        $exception = $request->attributes->get('exception');
+        if (($exception instanceof HttpExceptionInterface || $exception instanceof SymfonyHttpExceptionInterface) && $exceptionHeaders = $exception->getHeaders()) {
```

Comment:
I'm not a huge fan of this right now because of the HttpKernel dependency, I'll see to probably move this in the bridge in the future.